eventio/simtel/simtelfile.py

```diff
@@ -0,0 +1,394 @@
+"""
+    RunHeader[2000]
+    MCRunHeader[2001]
+    CORSIKAInputCard[1212]
+
+        # 1x per telescope (n_telescopes is in RunHeader)
+        # I call this TelescopeDescription
+    {
+        SimTelCamSettings[2002]
+        SimTelCamOrgan[2003]
+        SimTelPixelset[2004]
+        SimTelPixelDisable[2005]
+        SimTelCamsoftset[2006]
+        SimTelTrackSet[2008]
+        SimTelPointingCor[2007]
+    }
+
+    {
+        MCShower[2020](shower=3)
+        {
+            MCEvent[2021](event=301)
+            CORSIKATelescopeData[1204](event=301)
+                # IACTPhotoElectrons inside
+
+            { 1x per telescope and I don't know why they come here
+            TelMoni[2022](telescope_id=1, what=0x7f)
+            LasCal[2023](telescope_id=1)
+            }
+            MCPeSum[2026](id=301)
+            Event[2010]
+            {
+                CentEvent[2009](id=301)
+                TelEvent[2229](telescope_id=29, id=301)
+                {
+                    TelEvtHead[2011](telescope_id=29)
+                    TelADCSamp[2013](telescope_id=29,
+                    PixelTiming[2016](telescope_id=29)
+                    TelImage[2014](telescope_id=29,
+                    PixelList[2027](telescope_id=29
+                }
+                TelEvent[2237](telescope_id=37, id=301)
+                TrackEvent[2113](telescope_id=13
+                TrackEvent[2117](telescope_id=17
+                TrackEvent[2123](telescope_id=23
+                TrackEvent[2129](telescope_id=29
+                TrackEvent[2131](telescope_id=31
+                ...
+                TrackEvent[2163](telescope_id=63
+                Shower[2015]
+            }
+        }
+
+
+    }
+
+
+"""
+import logging
+from eventio.base import EventIOFile, EventIOObject
+
+class WrongType(Exception):
+    pass
+
+class NoTrackEvents(Exception):
+    pass
+
+class WithNextAssert:
```

Comment:
I think classes having this mixed in should also use `_last_obj` in the plain `__next__`, so users don't shoot themselvees in the foot...